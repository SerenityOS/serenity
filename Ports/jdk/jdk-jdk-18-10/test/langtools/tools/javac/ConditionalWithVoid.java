/*
 * @test /nodynamiccopyright/
 * @bug 4974927 8064464
 * @summary The compiler was allowing void types in its parsing of conditional expressions.
 * @author tball
 *
 * @compile/fail/ref=ConditionalWithVoid.out -XDrawDiagnostics ConditionalWithVoid.java
 */
public class ConditionalWithVoid {
    public void test(Object o, String s) {
        // Should fail to compile since Object.wait() has a void return type. Poly case.
        System.out.println(o instanceof String ? o.hashCode() : o.wait());
        // Should fail to compile since Object.wait() has a void return type. Standalone case.
        (o instanceof String ? o.hashCode() : o.wait()).toString();
        // Should fail to compile since Object.wait() has a void return type. Poly case.
        System.out.println(switch (s) {case "" -> o.hashCode(); default -> o.wait();});
        // Should fail to compile since Object.wait() has a void return type. Standalone case.
        (switch (s) {case "" -> o.hashCode(); default -> o.wait();}).toString();
    }
}
