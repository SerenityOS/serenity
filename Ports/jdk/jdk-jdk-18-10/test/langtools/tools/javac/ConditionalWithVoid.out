ConditionalWithVoid.java:12:71: compiler.err.void.not.allowed.here
ConditionalWithVoid.java:14:53: compiler.err.void.not.allowed.here
ConditionalWithVoid.java:16:82: compiler.err.void.not.allowed.here
ConditionalWithVoid.java:18:64: compiler.err.void.not.allowed.here
4 errors
